basket/base/rq.py

```diff
@@ -184,49 +183,59 @@ def store_task_exception_handler(job, *exc_info):
     # A job will retry if it has failed but has not yet reached the maximum number of retries.
     # A job is scheduled for retry when its status is `SCHEDULED`; otherwise, its status is set to `FAILED`.
 
-    if job._status == JobStatus.SCHEDULED:
-        # Job failed but is scheduled for a retry.
-        metrics.incr("base.tasks.retried", tags=[f"task:{task_name}"])
+    # Since there's no way to abort retries at the moment in RQ, we can set the `job.retries_left` to zero.
+    # This depends on the `handle_exception` call (which calls this code) to come before the
+    # `handle_job_failure` call which checks how many retries are left before calling `job.retry`.
+    # If `retries_left` is zero it goes to the FAILED state and the job doesn't get rescheduled.
 
-        if exc_info[1] not in EXCEPTIONS_ALLOW_RETRY:
-            # Force retries to abort.
-            # Since there's no way to abort retries at the moment in RQ, we can set the job `retries_left` to zero.
-            # This will retry one more time but no further retries will be performed.
+    if job._status == JobStatus.FAILED:
+        # Check if this is something we ignore and don't attempt to retry.
+        # If so, abort any retries, log to sentry, and return/skip the rest.
+        if ignore_error(exc_info[1]):
             job.retries_left = 0
+            sentry_capture(exc_info[1], "ignored")
+            return False
+
+        if job.retries_left and job.retries_left > 0:
+            # The job will be rescheduled for a retry.
+            if type(exc_info[1]) not in EXCEPTIONS_ALLOW_RETRY:
+                # If the job is not retryable, we abort any retries and consider this a task failure.
+                job.retries_left = 0
+                metrics.incr("base.tasks.failed", tags=[f"task:{task_name}"])
+                store_failed_task(job, *exc_info)
+                sentry_capture(exc_info[1], "failed")
+            else:
+                # Job failed, is retryable, and has retries left.
+                metrics.incr("base.tasks.retried", tags=[f"task:{task_name}"])
+                sentry_capture(exc_info[1], "retried")
+
+        else:
+            # Job failed and has no retries left.
+            metrics.incr("base.tasks.failed", tags=[f"task:{task_name}"])
+            store_failed_task(job, *exc_info)
+            sentry_capture(exc_info[1], "failed")
 
-        # Don't log to sentry if we explicitly raise `RetryTask`.
-        if not isinstance(exc_info[1], RetryTask):
-            with sentry_sdk.isolation_scope() as scope:
-                scope.set_tag("action", "retried")
-                sentry_sdk.capture_exception()
-
-    elif job._status == JobStatus.FAILED:
-        # Job failed but no retries left.
-        metrics.incr("base.tasks.failed", tags=[f"task:{task_name}"])
+    # Returning `False`` prevents any subsequent exception handlers from running.
```

Comment:
**nitpick (non-blocking):** Would be good to disclose this at the top - I was wondering why earlier we returned `False`